### ISSUE TITLE
"ANDROID_SUPPORT_VERSION is not defined" but it is in config.xml

### ISSUE THREAD

**Issue by pom421** [dpa99c/cordova-android-support-gradle-release#11]
I have a recurrent problem when the platforms directory is gone.

When i use "cordova prepare android", i got : 

```
Discovered platform "android@^6.3.0" in config.xml or package.json. Adding it to the project
Using cordova-fetch for cordova-android@^6.3.0
cordova-android-support-gradle-release  ANDROID_SUPPORT_VERSION:  26.+
Error: ANDROID_SUPPORT_VERSION is not defined
```
My package.json and config.xml are good, it seems : 

package.json

```
  "cordova": {
    "plugins": {
      "cordova-plugin-app-version": {},
      "cordova-plugin-device": {},
      "cordova-plugin-file": {},
      "cordova-plugin-file-opener2": {},
      "cordova-plugin-file-transfer": {},
      "cordova-plugin-inappbrowser": {},
      "cordova-plugin-network-information": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-whitelist": {},
      "cordova-plugin-dialogs": {},
      "com.bez4pieci.cookies": {},
      "cordova-plugin-clipboard2": {},
      "cordova-open-native-settings": {},
      "cordova-plugin-config-ios": {},
      "cordova.plugins.diagnostic": {},
      "phonegap-plugin-barcodescanner": {
        "CAMERA_USAGE_DESCRIPTION": "Need permission"
      },
      "cordova-android-support-gradle-release": {
        "ANDROID_SUPPORT_VERSION": "26.+"
      },

```

config.xml

```
    <plugin name="cordova-android-support-gradle-release" spec="^1.1.4">
        <variable name="ANDROID_SUPPORT_VERSION" value="26.+" />
    </plugin>

```

**Comment by pom421** [comment]
It seems to work when i type "cordova prepare" and to not work if i type "cordova prepare android". 
Strange...

**Comment by dpa99c** [comment]
> Error: ANDROID_SUPPORT_VERSION is not defined

This error is occurring due to an error in reporting an error!

If the hook script fails to find `platforms/android/build.gradle`, [it should log an error](https://github.com/dpa99c/cordova-android-support-gradle-release/blob/master/scripts/before-prepare.js#L47) and fail gracefully (i.e. not return an non-zero error code and therefore allow the outcome to be success).

However, there's a bug in that line of code:

    if (err) return console.log(PLUGIN_NAME, " FAILED TO WRITE ", GRADLE_FILENAME, " > ", ANDROID_SUPPORT_VERSION, err);
    
should be:

    if (err) return console.log(PLUGIN_NAME, " FAILED TO WRITE ", GRADLE_FILENAME, " > ", version, err);
    
Hence the error message `Error: ANDROID_SUPPORT_VERSION is not defined`.

I'll push a commit to fix this shortly.

FYI, the reason it works OK when you run `cordova prepare` is that the Cordova CLI will run hook scripts only for platforms which are currently present (i.e. none if `platforms/` is not present).

However, if you run `cordova prepare android`, the Cordova CLI will run the hook scripts for the Android platform regardless of whether the platform directory is present.

**Comment by pom421** [comment]
👍 Dave!
I will use the workaround in the meantime

**Comment by dpa99c** [comment]
Fix published to npm in v1.1.5
